examples/http/clearone_convergence.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Domotz Custom Driver 
+ * Name: ClearOne Converge
+ * Description: This script extracts information for ClearOne devices
+ * 
+ * Communication protocol is HTTP
+ * 
+ * Tested with ClearOne Converge Pro 2 
+ *
+ * Parameters: 
+ *    - urlCl: Url access to Convergence
+ *    - apiKey: Communicate with http api using API-X-KEY
+ *    
+ * Custom Driver Variables:
+ *    - Model: The model device
+ *    - Serial Number: The serial number device
+ *    - Firmware Version: Firmware version in device. (When * is shown means firmware pending update)
+ *    - Locate Mode: True when the device is mode locate, false otherwise
+ *    - Status: Status device (STATUS - TIME IN THIS STATUS)
+ *    - Location: Location device
+ *    - Part Number: Part number device
+ *
+ **/
+
+// Url access to Convergence
+var urlCl = D.getParameter('urlCl');
+
+// API Key access to Convergence
+var apiKey = D.getParameter('apiKey');
+var deviceInfo , detailInfo;
+
+/**
+ * @param {string} url  The URL to perform the GET request
+ * @returns promise for http response body
+ */
+function httpGet(url) {
+    var d = D.q.defer();
+	var website = D.createExternalDevice(urlCl);
+    var config = {
+        url: url,
+		protocol: "http:",
+		port: 8080,
+		headers: {
+			"X-Api-Key": apiKey,
+		},
+    };
+    website.http.get(config, function (error, response, body) {
+		if (error) {
+            console.error(error)
+            D.failure(D.errorType.GENERIC_ERROR);
+        }
+        if (response.statusCode == 404) {
+            D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+        }
+        if (response.statusCode === 401 || response.statusCode === 403) {
+            D.failure(D.errorType.AUTHENTICATION_ERROR);
+        }
+        if (response.statusCode != 200) {
+            console.error(error)
+            D.failure(D.errorType.GENERIC_ERROR);
+        }
+        d.resolve(JSON.parse(body));
+    });
+    return d.promise;
+}
+
+/**
+ * @returns promise for http response body containig device information
+ */
+function search() {
+    return httpGet("/dashboard-back/api/devices/pro-audio/search?q=" + D.device.ip())
+        .then(function(data){
+            deviceInfo = data.avDevices[0];
```

Comment:
what if the avDevices is not part of the data response? (eg an error or no avDevices in the response body at all)